internal/pkg/levee/levee.go

```diff
@@ -18,22 +18,28 @@ import (
 	"fmt"
 	"strings"
 
+	"github.com/google/go-flow-levee/internal/pkg/utils"
+
 	"github.com/google/go-flow-levee/internal/pkg/call"
 	"github.com/google/go-flow-levee/internal/pkg/config"
 	"github.com/google/go-flow-levee/internal/pkg/fieldpropagator"
 	"github.com/google/go-flow-levee/internal/pkg/varargs"
 	"golang.org/x/tools/go/analysis"
+	"golang.org/x/tools/go/analysis/passes/buildssa"
+	"golang.org/x/tools/go/pointer"
 	"golang.org/x/tools/go/ssa"
 
 	"github.com/google/go-flow-levee/internal/pkg/source"
 )
 
+const sourceReachedSinkMessage = "a source has reached a sink"
```

Comment:
I think you make a valid point. As I understand it, the only difference between the two kinds of report is the *location*, i.e. at the source or at the sink. In both cases we are reporting on the same kind of event, i.e. that "a source has reached a sink". This difference is already present in the messages: `, source: %v` vs `, sink: %v`, but that may not be obvious because the reporting functions' bodies are largely the same. Do you think a refactoring is needed?

I've updated the `want` messages in `propagators` to show what kind of message we expect.

Reply:
> This difference is already present in the messages: , source: %v vs , sink: %v, but that may not be obvious because the reporting functions' bodies are largely the same. Do you think a refactoring is needed?

It was more of an issue of testing readability than reporting.  Since `// want "text"` checks `text` as a regexp, and since our `// want` lines aren't anchored and we hadn't included the appended suffix.  Including it in the `// want` as you have suffices, though.